src/Internal/QueryM.purs

```diff
@@ -538,32 +550,79 @@ allowError func = func <<< Right
 -- Wallet
 --------------------------------------------------------------------------------
 
-getWalletAddresses :: QueryM (Maybe (Array Address))
-getWalletAddresses = do
+getNetworkId :: QueryM Int
+getNetworkId = do
+  maybeWallet <- asks $ _.wallet <<< _.runtime
+  configId <- asks $ _.networkId <<< _.config
+  liftAff case maybeWallet of
+    Just (Eternl wallet) -> callCip30Wallet wallet _.getNetworkId
+    Just (Nami wallet) -> callCip30Wallet wallet _.getNetworkId
+    Just (Gero wallet) -> callCip30Wallet wallet _.getNetworkId
+    Just (Flint wallet) -> callCip30Wallet wallet _.getNetworkId
+    Just (Lode wallet) -> callCip30Wallet wallet _.getNetworkId
+    Just (KeyWallet _) -> pure $ case configId of
+      TestnetId -> 0
+      MainnetId -> 1
+    Nothing -> liftEffect $ throw "Can't get Wallet from config"
```

Comment:
I don't think we should have this. We already have `getNetworkId` in `QueryM.NetworkId`, which is the one exported in `Contract.Address`. I think we should immediately fail during `mkQueryRuntime` by doing the cip30 call there after the wallet is initialized via the spec.